File: smartPlantAngular/src/app/Models/SmartPot.ts
import {Plant} from "./Plant";
import {UserSmartPot} from "./UserSmartPot";
export class SmartPot {
  get plant(): Plant {
    return this._plant;
  }

  set plant(value: Plant) {
    this._plant = value;
  }
  get userSmartPot(): UserSmartPot[] {
    return this._userSmartPot;
  }

  set userSmartPot(value: UserSmartPot[]) {
    this._userSmartPot = value;
  }
  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }
  get MACAddress(): string {
    return this._MACAddress;
  }

  set MACAddress(value: string) {
    this._MACAddress = value;
  }
  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }
  private _id!: number;
  private _MACAddress!: string;
  private _name!: string;
  private _userSmartPot!: UserSmartPot[];
  private _plant!: Plant;
}

File: smartPlantSpringBoot/_/src/main/java/com/app/tddt4iots/entities/SmartPot.java
package com.app.tddt4iots.entities;

import com.app.tddt4iots.enums.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.*;
import java.lang.*;

import jakarta.persistence.*;
import java.util.List;

@Entity
@Table(name = "SmartPot")
@Data
@NoArgsConstructor
public class SmartPot {
    
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
     
    @Column(name = "MACAddress", nullable = true, unique = true, length = 30) 
    private String MACAddress; 
 
    @Column(name = "name", nullable = false, unique = false, length = 30) 
    private String name; 
 
    @OneToMany(mappedBy = "id") 
    private List<UserSmartPot> userSmartPot; 
 
    @JoinColumn(name = "idPlant", referencedColumnName = "id") 
    @OneToOne 
    private Plant plant; 
 
    
}    
